build-system/compile/compile.js

```diff
@@ -400,10 +401,13 @@ function compile(
         .on('error', reject)
         .pipe(sourcemaps.write('.'))
         .pipe(
-          gulpIf(
-            !!argv.esm,
-            gap.appendText(`\n//# sourceMappingURL=${outputFilename}.map`)
-          )
+          gulpIf(function(file) {
+            // Do not append sourceMappingURL if its a sourcemap
+            return (
+              pathModule.extname(file.path) !== '.map' &&
+              options.esmPassCompilation
+            );
+          }, gap.appendText(`\n//# sourceMappingURL=${outputFilename}.map`))
```

Comment:
Can you move out the condition function and the processing function into helpers called `shouldAppendSourceMappingUrl()` and `appendSourceMappingUrl()`?

Reply:
done.